docker/gcp_pudl_etl.sh

```diff
@@ -100,60 +95,95 @@ function notify_slack() {
 }
 
 function update_nightly_branch() {
-    git config --unset http.https://github.com/.extraheader
-    git config user.email "pudl@catalyst.coop"
-    git config user.name "pudlbot"
-    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git"
-    echo "BOGUS: Updating nightly branch to point at $NIGHTLY_TAG."
-    git fetch origin nightly:nightly
-    git checkout nightly
-    git merge --ff-only "$NIGHTLY_TAG"
-    ETL_SUCCESS=${PIPESTATUS[0]}
-    git push -u origin
+    git config --unset http.https://github.com/.extraheader && \
+    git config user.email "pudl@catalyst.coop" && \
+    git config user.name "pudlbot" && \
+    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git" && \
+    echo "Updating nightly branch to point at $NIGHTLY_TAG." && \
+    git tag -d "$NIGHTLY_TAG" && \
+    git fetch --tags origin "$NIGHTLY_TAG" && \
+    git checkout "$NIGHTLY_TAG" && \
+    git tag && \
+    git fetch origin nightly:nightly && \
+    git checkout nightly && \
+    git merge --ff-only "$NIGHTLY_TAG^0" && \
+    git push -u origin nightly
```

Comment:
```suggestion
    git fetch --force --tags origin "$NIGHTLY_TAG" && \
    git fetch origin nightly:nightly && \
    git checkout nightly && \
    git show-ref -d nightly "$NIGHTLY_TAG" && \
    git merge --ff-only "$NIGHTLY_TAG" && \
    git push -u origin nightly
```
* use `--force` flag on git fetch to overwrite any local version of the $NIGHTLY_TAG, instead of manually deleting
* I think you don't have to check out the tag to be able to merge it later
* `show-ref -d` shows us "which commit hashes are we pointing at", which is both more useful & less verbose than `git tag`
* not sure why the diff isn't rendering nicely, but the merge/push and the nightly branch fetch/checkout all seem fine.